src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedConditionalOperatorTests.cs

```diff
@@ -416,5 +416,42 @@ static void Main()
                 .VerifyDiagnostics();
             CompileAndVerify(comp, expectedOutput: expectedOutput);
         }
+
+        [Fact, WorkItem(46231, "https://github.com/dotnet/roslyn/issues/46231")]
+        public void TestFixedConditional()
+        {
+            var source = @"
+public class Program {
+    public unsafe static void Test(bool b, int i)
+    {
+        fixed (byte * p = b ? new byte[i] : null)
+        {
+        }
+    }
+}";
+            CreateCompilation(source, parseOptions: TestOptions.Regular8, options: TestOptions.DebugDll.WithAllowUnsafe(true))
+                .VerifyEmitDiagnostics();
+            CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(MessageID.IDS_FeatureTargetTypedConditional.RequiredVersion()), options: TestOptions.DebugDll.WithAllowUnsafe(true))
+                .VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void TestUsingConditional()
+        {
+            var source = @"
+using System;
+public class Program {
+    public static void Test(bool b, IDisposable d)
+    {
+        using (IDisposable x = b ? d : null)
```

Comment:
Does the implementation change affect the behavior of this test?

Reply:
Yes. It causes the compiler to not crash. Other than that, no.

Reply:
I was not able to crash the compiler in the master branch using this scenario. I was able to crash it using the `TestFixedConditional` scenario above. This seems like a good test to have anyway, though.

Reply:
You're right; this test is just a defensive one for a "similar" situation to the one that is actually affected here.